pkgs/servers/http/h2o/default.nix

```diff
@@ -14,6 +14,15 @@ stdenv.mkDerivation rec {
     sha256 = "0lwg5sfsr7fw7cfy0hrhadgixm35b5cgcvlhwhbk89j72y1bqi6n";
   };
 
+  patches = [
+    (substituteAll rec {
+      src = ./nix-store-etag.patch;
+      nixStoreDir = builtins.storeDir;
+      nixStoreHashLen = with stdenv.lib;
+        stringLength (head (builtins.split "-" (baseNameOf "${src}")));
```

Comment:
I didn't realize this counted as an import-from-derivation :(

I'd like to avoid it if possible; I'm happy to just hard-code `32`, but would prefer a nicer way to get the store hash length in a future-proof manner. Unfortunately, I can't think of one...

Reply:
What about `strchr` towards `-` at runtime like we do in the nginx patch?

Reply:
Doesn't look like IFD to me. Isn't this pretty much the same as `stdenv.mkDerivation { /* ... */ buildPhase = "${coreutils}/bin/ls"; }`?

Reply:
> [...] so it won't build in read-only mode.

Hm, did you mean that it won't evaluate? `nix-instantiate --readonly-mode --eval -A h2o` works, nothing will build in read-only mode by definition.

Reply:
@yegortimoshenko: Correct, I meant eval here.

And yes, @emilazy also: I was wrong in regards to IFD: Since the store path doesn't actually *need* to exist for evaluation, it isn't an IFD (also, there is no `import` involved, duh!). Sorry for the noise.

Reply:
@aszlig Note that IFD is a bit of a misnomer, because the same effect can happen with a `builtins.readFile <derivation>` (notably without an `import`). A better name for IFD would be "derivation-dependent evaluation"